Explorer/Assets/Plugins/CommandTerminal/ConsoleBuffer.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+// ORIGINAL OPEN SOURCE PLUGIN SCRIPT:
+// https://github.com/stillwwater/command_terminal/blob/0f5918ea79014955b24c7d431625a97a1cac8797/CommandTerminal/CommandLog.cs
+
+namespace CommandTerminal
+{
+    public struct LogItem
+    {
+        public LogType type;
+        public string message;
+        public string stackTrace;
+    }
+
+    public class ConsoleBuffer
+    {
+        private readonly int maxItems;
+
+        public List<LogItem> Logs { get; } = new ();
+
+        public ConsoleBuffer(int maxItems)
+        {
+            this.maxItems = maxItems;
+        }
+
+        public void HandleLog(string message, string stackTrace, LogType logType)
+        {
+            var log = new LogItem
+            {
+                message = message,
+                stackTrace = stackTrace,
+                type = logType,
+            };
+
+            Logs.Add(log);
+
+            if (Logs.Count > maxItems) { Logs.RemoveAt(0); }
```

Comment:
O(n) complexity on the condition, I assume logs list shouldn't be too big (512 elements), but how often it could be called? If it is then we could consider other approach

Reply:
Shouldn't be often at all, only when the scene triggers a log. Also this only runs during local scene development, without having any other scene than the local one using all the performance for itself.
We can change this in the future if we identify a problem by then ;)

Reply:
Added comment at https://github.com/decentraland/unity-explorer/pull/1546/commits/7db392aa7c131317dd50483f83fe60edbcb8d882